.github/workflows/maven.yml

```diff
@@ -49,7 +49,6 @@ jobs:
       - name: Analyse with SonarCloud
         run: >
           ./mvnw --batch-mode
-          -Pcoverage
```

Comment:
We need the coverage profile here to generate correct sonarcloud analyses

Reply:
Did you check the commit message (body)?

https://github.com/synyx/urlaubsverwaltung/pull/1872/commits/95b3e90ec147079d6dc11f5b3abd7f489f393c78

>Collecting coverage information is only needed during test execution, i.e. in the previous step.

You run your tests in the previous step (`verify`/`install`), this is where you already collect the coverage data via JaCoCo:

https://github.com/synyx/urlaubsverwaltung/blob/9e478635ac6595367bdf31e866a9d5021200075d/.github/workflows/maven.yml#L43

When you trigger the Sonar goal `sonar:sonar`, you no longer need that profile.

Reply:
No I did not :-/ But is it save to remove this and use the correct/same version of the sonarsource plugin or does maven maybe take another one if it is not specified via the profile? I have not tested that. But this is a good hint. Maybe we do not need this.

Reply:
If I add the sonar plugin to the global plugin structure
and run it I get

```shell
➜  urlaubsverwaltung git:(master) ./mvnw sonar:sonar          
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< org.synyx:urlaubsverwaltung >---------------------
[INFO] Building Urlaubsverwaltung 4.13.0-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- sonar-maven-plugin:3.7.0.1746:sonar (default-cli) @ urlaubsverwaltung ---
...
```

```shell
➜  urlaubsverwaltung git:(master) ✗ ./mvnw -Pcoverage sonar:sonar
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< org.synyx:urlaubsverwaltung >---------------------
[INFO] Building Urlaubsverwaltung 4.13.0-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- sonar-maven-plugin:3.8.0.2131:sonar (default-cli) @ urlaubsverwaltung ---
...
```

so if we do not specifiy the profile it is not given that we use the same version of the plugin in every case?

Reply:
Does maven analyse the plugin and goal given and download the latest version of it? I should take a look in the docs.

Reply:
Wir können eigentlich auch deutsch reden, oder? 😉

Grundsätzlich ist es eine gute Idee, die Version den Maven SonarScanner Plugins explizit zu setzen (siehe ["How to Fix Version of Maven Plugin"](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-maven/)).

Mit dem `coverage`-Profil erzeugst du lediglich die Coverage-Informationen, welche sich anschließend unter `target/site/jacoco/` finden – das genügt `sonar:sonar`. Aus ["Analyzing"](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-maven/):

>To get coverage information, you'll need to generate the coverage report _before_ the analysis.

So mache ich es zumindest in meinen Projekten, Bsp.: [PDF Zoom Wizard](https://github.com/beatngu13/pdf-zoom-wizard). Und [hier](https://sonarcloud.io/dashboard?id=com.github.beatngu13%3Apdfzoomwizard) der zugehörige SonarCloud-Status.

BTW wenn du beim Coverage auch die ITs berücksichtigen möchtest, kannst du das wie folgt tun:

https://github.com/beatngu13/pdf-zoom-wizard/blob/develop/pom.xml#L133

Wenn du magst, committe ich das noch sowie die feste Plugin-Version (bei Bedarf in einem neuen PR)?

Reply:
Lass uns das Profil hier noch drin lassen und auf eine feste Version gehen, ich würde hier noch ungern in unterschiedliche Versionen laufen. Das mit den IT-Tests schaue ich mir mal die Tage an. Ich bin mir nicht mehr sicher warum wir das nicht gemacht hatten. Muss ich mal genauer anschauen